lib/src/parse/parser.dart

```diff
@@ -517,24 +517,33 @@ class Parser {
     return true;
   }
 
-  /// Consumes the next character if it's equal to [letter], ignoring ASCII
-  /// case.
+  /// Consumes the next character or escape sequence if it matches [letter].
+  ///
+  /// [letter] must be a lowercase ASCII character. It will be matched
+  /// case-insensitively unless [caseSensitive] is true.
   @protected
-  bool scanCharIgnoreCase(int letter) {
-    if (!equalsLetterIgnoreCase(letter, scanner.peekChar())) return false;
-    scanner.readChar();
-    return true;
+  bool scanLetter(int letter, {bool caseSensitive = false}) {
+    var start = scanner.state;
```

Comment:
Storing the state for every letter could get expensive, since it involves an allocation of a state object, and parse speed can be important for a lot of ancillary tools like static analyzers. It may be worth only storing this if we actually see a backslash in practice, which is likely to be very rare.